Add one colourscheme recipe

balajisivaraman opened the pull request:
### Brief summary of what the package does

This package is port of the [Vim One](https://github.com/rakr/vim-one/) colourscheme to Emacs. It provides both the Light and Dark variants of the same.

### Direct link to the package repository

https://github.com/balajisivaraman/emacs-one-themes

### Your association with the package

I'm the maintainer of this package.

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

DamienCassou commented:
Some feedback:

- please add `-*- lexical-binding: t -*-` at the end of the first line
- you may want to specify the minimum Emacs requirement with a dedicated line:

    ```emacs
    ;; Package-Requires: ((emacs "24"))
    ```
- please provide docstrings for public variables and constants
- you shouldn't quote argument names in docstrings (e.g., in `emacs-one-with-colour-variables`)
- usually, code is in American English (e.g., "color") instead of British English (e.g., "colour")
- please fix indentation of the files
- the linters are unhappy with the two theme files. Please fix.

purcell commented:
Thanks. More feedback from me:

- `emacs-one.el` doesn't fit with the other files. In a package with the name `one-theme`, all the files should match the pattern `one-theme*.el`.
- The main file (the elisp file whose name matches that of the package) must contain the standard package boilerplate header and relevant metadata.
- `one-theme.el` will be found on the theme load path, and listed as a duplicate theme.

Overall, I suggest you do the following:

1. Put the contents of `emacs-one.el` into `one-themes.el` (note the "s"). Rename all identifiers in that file so they have the `one-themes-` prefix. Loading that file on its own should not do anything.
2. Make `one-light-theme.el` and `one-dark-theme.el` use the functions in that file.
3. Rename the recipe to `one-themes`.

balajisivaraman commented:
@DamienCassou & @purcell, Thank you for the feedback. I will work on them and update the PR in the coming days.

balajisivaraman commented:
I have addressed the feedback in my repository and, hopefully, now everything is in order.

Please note that, I have not made any changes only for this feedback: **"the linters are unhappy with the two theme files. Please fix."**. This is because I took inspiration from Purcell's Tomorrow themes and BBatsov's Solarized themes, both of which have very minimal variant files. If this is absolutely necessary, I can go ahead and address this as well. Please let me know.

tarsius commented:
@purcell, @DamienCassou since you reviewed earlier, does this look okay now?

DamienCassou commented:
My comments have been addressed.

purcell commented:
Great, thanks - welcome to MELPA! :tada:

balajisivaraman commented:
Awesome! Thank you all for the review comments and for merging. I feel like I'm part of an exclusive club now. :-)

purcell commented:
> I feel like I'm part of an exclusive club now. :-)

You absolutely are: the unusual club composed of Emacs enthusiasts!
